src/main/kotlin/sc/gui/controller/GameController.kt

```diff
@@ -198,6 +198,10 @@ class GameController : Controller() {
         }
         subscribe<HumanMoveRequest> { event ->
             logger.debug("Human move request")
+
+            val moves = GameRuleLogic.getPossibleMoves(event.gameState)
```

Comment:
Could we reduce that to a set of possible shapes here, so that the check below doesn't have to go through all moves for every shape?

Reply:
yeah, I did that meanwhile.
Still, had to wait for the other PR